### PR TITLE
Some improvements to the redirect handling

### PR DESCRIPTION
**Pull request by cameronterry**
This PR transposes some of the mechanisms from upcoming changes in my own domain mapping plugin that seemed equally relevant here. The changes include:

* ~Switched the handler to `set_404` hook rather than `template_redirect` - it fires earlier in the process and only fires when `WP_Query` determines there is a 404. **Note:** this means the plugin code only supports WordPress 5.5.x or newer.~ Switched it back to `template_redirect` for broader compatibility between older and newer versions of WordPress; https://github.com/svandragt/404-fallback/pull/9/commits/46a67be8ba535c8b6fb81033ff5eb28e369466e1.
* Enhanced validation for the Fallback URL during the redirect phase, to ensure the admin controls cannot be circumvented with say the WP CLI using `wp option update fb404_setting_fallback_url "gibberish"`. This fixes half of the need for https://github.com/svandragt/404-fallback/issues/8.
* Sub-folder multisite support for Request. So in setups which have, https://multisite.test/sub-folder/, it will now ensure `/sub-folder/` doesn't get prepended to the Request on the Fallback URL. Most likely a very rare edge case, as I imagine most sites using this will be mapped in some form.
* Made sure the Fallback URL + Request URI join doesn't accidentally get a double slash `//`.
* Adding some attribute escaping and reworked in the Fallback URL field.
  * Type is now `url` which will now benefit from in-browser (for modern browsers at least ...) validation.
  * Set the `required` attribute to benefit from in-browser validation.
  * Ensured the value attribute has is escaped.
  * Wrapped the `<input>` in `wp_kses()`. Probably overkill but since it was a string and this guarantees the HTML output, it's a bit safer.
  * This partly improves https://github.com/svandragt/404-fallback/issues/8 but the sanitisation will need to be added to [Settings API implementation here](https://github.com/svandragt/404-fallback/blob/ba44e4089558163c3c97735b644a92b07ae8aa47/lib/menus.php#L49-L50). You could reuse the `fb404_validate_url()` function I've added for this. **Note:** I left this out as because there is a few issues around Menu that looked better suited for solving this side.

### CODE REVIEW

File: 404-fallback.php [hunk 1]
```diff
@@ -17,20 +17,26 @@
...
+	/**
+	 * We make sure this URL has a trailing slash, as we removed it from the Request.
+	 */
+	$location  = trailingslashit( $url);
+	$location .= $request;
+
+	$status        = 302;
+	$x_redirect_by = '404 Fallback';
+
+	wp_redirect( $location, $status, $x_redirect_by );
```
Comment (cameronterry) [inline 404-fallback.php:36]: Just noting; I'd be tempted to wrap `$location` in an `apply_filters()` to enable some customisation.

Main use case I can think of is wanting to add like `utm_source` for analytics and logs on the receiver. Could be useful for some to be able to break down how many requests are coming through the Fallback URL configuration.

**Review by svandragt — APPROVED** [review]
Looks good, will put it in main.
Wanted to merge the improvements before I'm going to archive the repo.
